lib/ui/Popup.js

```diff
@@ -199,13 +200,13 @@ Popup.prototype = {
         var imgMatches = chromeImgLink.match(/^moz-anno\:favicon\:(.*)/);
 
         results.push({
-          url: Components.classes["@mozilla.org/intl/texttosuburi;1"].
+          url: Components.classes['@mozilla.org/intl/texttosuburi;1'].
                 getService(Components.interfaces.nsITextToSubURI).
-                unEscapeURIForUI("UTF-8", controller.getValueAt(i)),
+                unEscapeURIForUI('UTF-8', controller.getValueAt(i)),
           image: imgMatches ? imgMatches[1] : null,
           title: controller.getCommentAt(i),
           type: controller.getStyleAt(i),
-          text: controller.searchString.replace(/^\s+/, "").replace(/\s+$/, "")
+          text: controller.searchString.replace(/^\s+/, '').replace(/\s+$/, '')
```

Comment:
This code comes to us from the [sick, sad world of XBL](https://github.com/mozilla/gecko-dev/blob/master/toolkit/content/widgets/autocomplete.xml#L1156-L1157), and in fact those lines were originally created before `String.prototype.trim` was a thing (added to FF in version 3.5).

So yeah, I should fix that. Or you're welcome to.